### ISSUE TITLE
EventGhost fails to start due to missing stackless module

### ISSUE THREAD

**Issue by per1234** [EventGhost/EventGhost#201]
Operating system: Windows 7  64-bit and 32-bit
EventGhost version: build [438](https://ci.appveyor.com/api/buildjobs/hu0sauq213u81bcu/artifacts/_build%2Foutput%2FEventGhost_WIP-2017.02.15-17.26.06_Setup.exe), [440](https://ci.appveyor.com/api/buildjobs/qtis9jw41hq27x8u/artifacts/_build%2Foutput%2FEventGhost_WIP-2017.02.17-16.18.02_Setup.exe), [445](https://ci.appveyor.com/api/buildjobs/fotwveyaisgksckw/artifacts/_build%2Foutput%2FEventGhost_WIP-2017.02.21-17.42.30_Setup.exe)

EventGhost fails to start with any build newer than [430](https://ci.appveyor.com/api/buildjobs/nwsymqe4u1j3nhrp/artifacts/_build%2Foutput%2FEventGhost_WIP-2017.02.12-18.48.20_Setup.exe). The output shown in Log.txt (on the 64-bit system) is:
```
Traceback (most recent call last):
  File "EventGhost.pyw", line 53, in <module>
  File "C:\Program Files (x86)\EventGhost\eg\__init__.py", line 19, in <module>
    import stackless
  File "stackless.pyc", line 8, in <module>
ImportError: No module named _stackless
```
The issue appears to have been caused by https://github.com/EventGhost/EventGhost/commit/9deea71fbab1083158dc7e67cd966cfaecf62ab0 because the issue occurs after that commit.

CC: @kdschlosser

**Comment by kdschlosser** [comment]
I will take a look at the appveyor setup on it and see what is going on. it's a bit hairy with appveyor because it setup up the build environment with python installed already and then we install stackless I do not remember if i tried to cache the stackless installation or not.. and if so i will have to remove that bit of it and have it install it every single time.

**Comment by topic2k** [comment]
The install of Stackless Python fails because the default python installation exe is in use:
```
Info 1603.The file C:\Python27\python.exe is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\bz2.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\pyexpat.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\select.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\unicodedata.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\_elementtree.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\_socket.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\_ssl.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\_ctypes.pyd is being held in use.  Close that application and retry.
Info 1603.The file C:\Python27\DLLs\_hashlib.pyd is being held in use.  Close that application and retry.
MSI (s) (44:98) [20:55:26:911]: Setting cached product context: User non-assigned for product: 5EC82AD95AA0E924688D86E68D896CD6
Action ended 20:55:27: InstallFinalize. Return value 1.
Action ended 20:55:27: INSTALL. Return value 1.
MSI (s) (44:98) [20:55:27:193]: Product: Python 2.7.12150 (Stackless) -- Installation completed successfully.

MSI (s) (44:98) [20:55:27:193]: Windows Installer installed the product. Product Name: Python 2.7.12150 (Stackless). Product Version: 2.7.12150. Product Language: 1033. Manufacturer: The Stackless Team. Installation success or error status: 0.

MSI (s) (44:98) [20:55:27:193]: Windows Installer requires a system restart. Product Name: Python 2.7.12150 (Stackless). Product Version: 2.7.12150. Product Language: 1033. Manufacturer: The Stackless Team. Type of System Restart: 2. Reason for Restart: 1.

MSI (s) (44:98) [20:55:27:193]: Product: Python 2.7.12150 (Stackless). Restart required. The installation or update for the product required a restart for all changes to take effect.  The restart was deferred to a later time.
```

I don't know why python is running at that time. When i add a `- cmd: taskkill /F /IM python.exe` before the power shell script, it says python.exe is not running.

**Comment by kdschlosser** [comment]
that is strange

**Comment by kdschlosser** [comment]
i wonder if the actual unzipping of the cache is causing the problem. maybe the program that unzips it isn't being closed?

**Comment by kdschlosser** [comment]
and another question is.... how is it passing the dependency check if stackless is not actually being installed??

**Comment by kdschlosser** [comment]
and I just noticed this as well...
the version installed is 2.7.12150
and the check dependency version is this 2.7.9150
I know it checks to see which is greater but we probably should have those be the same. yes?

**Comment by kdschlosser** [comment]
ok I found an issue. stackless python is python 2.7.12 and the installed python version on appveyor 2.7.13. this could be an issue. I do not know when appveyor started using 2.7.13 but by default it is set to that installation. and there is currently no other python versions installed. and it defaults to the python27 directory for install just like stackless does. so i can see this bumping heads possibly. 

I am going to see if we can change the default folder for the stackless install from a switch using the msi file. if so this will also provide a nice means to cache the whole thing and not have to worry about caching any of the downloaded files.

I know that i am unable to install stackless from pip or easy_install because of the python version being 2.7.13 I think this may be the way to go around it. and could prove to be a much better solution in the end anyways

**Comment by kdschlosser** [comment]
well i have been working with this issue some more. and the method i have come up with i really do not like. But because of stackless python not having things like pip and setuptools installed makes it a rather large pain. I have found the solution to get it to install into a different directory. so I may have to install python and then get stackless to install over it.

**Comment by kdschlosser** [comment]
well I have found out that if i install stackless over python 2.7.13 locally i get the exact same error.

**Comment by kdschlosser** [comment]
well I know what the problem is... and even if i install stackless to a different directory the problem still occurs.. 

I am not sure what is causing it. but the python27.dll that gets included into the installer is the wrong one. if i install EG and then replace it from a working build. it runs fine. I do not know how this file gets made but it is not in the stackless python installation nor is it in the eventghost source. so maybe someone could shed some light as to where it comes from.. No matter what version of stackless i install i get the same error. I am thinking that stackless build the file at runtime maybe.. not sure. i do know this tho that file is included in the pip installation of stackless. but because of the pip installed version of stackless and it's use of a zip file py2exe is incompatible with this and is unable to grab the files..

But on a good note I have come up with a much more efficient appveyor..

**Comment by topic2k** [comment]
Stackless Python installs the python27.dll into c:\Python27, but there is also python27.dll in windows directory from the appveyor default python installation. Just delete those and everything is fine.

**Comment by kdschlosser** [comment]
YAY it works!! and I have also managed to cut the build time down to 1 minute and 55 seconds.. w00t.

I will issue a PR in a minute on this. so get ready to stamp this CLOSED!!

**Comment by kdschlosser** [comment]
also I engineered the newest appveyor script to not deal with anything that Appveyor does. so if they make any changes again to the python version it shouldn't (i hope) screw up our build process.

**Comment by topic2k** [comment]
fixed by #209
